sources/multi.nhentai/src/models.rs

```diff
@@ -0,0 +1,170 @@
+use aidoku::{
+	alloc::{string::String, string::ToString, Vec},
+	prelude::*,
+	ContentRating, Manga, MangaStatus, Viewer,
+};
+use serde::{Deserialize, Serialize};
+
+use crate::settings::{get_title_preference, TitlePreference};
+
+pub fn extension_from_type(t: &str) -> &str {
+	match t {
+		"j" => "jpg",
+		"p" => "png",
+		"w" => "webp",
+		"g" => "gif",
+		_ => "jpg", // default to jpg
+	}
+}
+
+#[derive(Deserialize, Serialize, Debug, Clone)]
+pub struct NhentaiTag {
+	pub id: i32,
+	pub name: String,
+	pub count: i32,
+	pub r#type: String,
+	pub url: String,
+}
+
+#[derive(Deserialize, Serialize, Debug, Clone)]
+pub struct NhentaiImage {
+	pub t: String,
+	pub w: i32,
+	pub h: i32,
+}
+
+#[derive(Deserialize, Serialize, Debug, Clone)]
+pub struct NhentaiImages {
+	pub pages: Vec<NhentaiImage>,
+	pub cover: NhentaiImage,
+	pub thumbnail: NhentaiImage,
+}
+
+#[derive(Deserialize, Serialize, Debug, Clone)]
+pub struct NhentaiGallery {
+	pub id: i32,
```

Comment:
I think this sometimes breaks, since the api returns some titles with a string id instead of a number. you can use the serde_json `Value` type instead.
```suggestion
	pub id: Value,
```
and above,
```rs
use serde_json::Value;
```

Reply:
Done